clang/test/SemaCXX/offsetof.cpp

```diff
@@ -17,7 +17,7 @@ struct Base { int x; };
 struct Derived : Base { int y; };
 int o = __builtin_offsetof(Derived, x); // expected-warning{{offset of on non-POD type}}
 
-const int o2 = sizeof(__builtin_offsetof(Derived, x));
+const int o2 = sizeof(__builtin_offsetof(Derived, x)); // expected-warning{{offset of on non-POD type 'Derived'}}
```

Comment:
This is an example of why I think the existing behavior is correct. The fact that we're asking for the offset of a field on a non-POD type is not harmful because the call is never evaluated.

Reply:
So you think we should ie, check for `!isUnevaluated()` - ie diag in constant expressions but not in unevaluated contexts?

Reply:
I think that might produce slightly better results by silencing this kind of warning.

Reply:
Fixed. Thanks.